browser/ui/webui/settings/brave_email_aliases_handler.cc

```diff
@@ -0,0 +1,447 @@
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.updateAlias",
+      base::BindRepeating(&BraveEmailAliasesHandler::UpdateAlias,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.deleteAlias",
+      base::BindRepeating(&BraveEmailAliasesHandler::DeleteAlias,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.requestAccount",
+      base::BindRepeating(&BraveEmailAliasesHandler::RequestAccount,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.getSession",
+      base::BindRepeating(&BraveEmailAliasesHandler::GetSession,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.getAccountEmail",
+      base::BindRepeating(&BraveEmailAliasesHandler::GetAccountEmail,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.logout",
+      base::BindRepeating(&BraveEmailAliasesHandler::Logout,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.closeBubble",
+      base::BindRepeating(&BraveEmailAliasesHandler::CloseBubble,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.fillField",
+      base::BindRepeating(&BraveEmailAliasesHandler::FillField,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "email_aliases.showSettingsPage",
+      base::BindRepeating(&BraveEmailAliasesHandler::ShowSettingsPage,
+                          base::Unretained(this)));
+}
+
+void BraveEmailAliasesHandler::SetNote(const std::string& alias_email,
+                                       const std::string& note) {
+  ScopedDictPrefUpdate update(GetProfile()->GetPrefs(), kEmailAliasesNotes);
+  update->Set(alias_email, note);
+}
+
+void BraveEmailAliasesHandler::DeleteNote(const std::string& alias_email) {
+  ScopedDictPrefUpdate update(GetProfile()->GetPrefs(), kEmailAliasesNotes);
+  update->Remove(alias_email);
+}
+
+std::optional<std::string> BraveEmailAliasesHandler::GetNote(
+    const std::string& alias_email) {
+  const auto* note = GetProfile()
+                         ->GetPrefs()
+                         ->GetDict(kEmailAliasesNotes)
+                         .FindString(alias_email);
+  return note ? std::optional<std::string>(*note) : std::nullopt;
+}
+
+void BraveEmailAliasesHandler::ApiFetch(
+    const GURL& url,
+    const char* method,
+    const std::optional<std::string>& bearer_token,
+    const base::Value::Dict& bodyValue,
+    network::SimpleURLLoader::BodyAsStringCallback
+        download_to_string_callback) {
+  auto resource_request = std::make_unique<network::ResourceRequest>();
```

Comment:
this kind of logic does not belong in the webui. It seems like there should be a keyed service to manage this stuff and that can also act as the mojo interface for the webui.

Reply:
@kdenhartog should we have
```
request->referrer_policy = net::ReferrerPolicy::NO_REFERRER;
  request->credentials_mode = network::mojom::CredentialsMode::kOmit;
``` here?